pkg/extender/extender.go

```diff
@@ -273,15 +275,15 @@ func (e *Extender) processFilterRequest(w http.ResponseWriter, req *http.Request
 			}
 
 			hyperconvergenceVolumeCount := 0
-			nodeHyperconverenceVolumeCount := make(map[string]int)
+			nodeHyperconvergenceVolumeCount := make(map[string]int)
 			for _, volumeInfo := range driverVolumes {
 				if !volumeInfo.NeedsAntiHyperconvergence {
 					hyperconvergenceVolumeCount++
 				}
 				for _, volumeNode := range volumeInfo.DataNodes {
-					// nodeToHyperconverenceVolumeCount is used to determine if valid nodes exist with preferLocalNodeOnly
+					// nodeToHyperconvergenceVolumeCount is used to determine if valid nodes exist with preferLocalNodeOnly
 					if preferLocalOnly && !volumeInfo.NeedsAntiHyperconvergence {
-						nodeHyperconverenceVolumeCount[volumeNode]++
+						nodeHyperconvergenceVolumeCount[volumeNode]++
 					}
```

Comment:
I think putting an additional comment in this loop (especially near nodeHyperconvergenceVolumeCount or !volumeInfo.NeedsAntiHyperconvergence and) would be helpful to understand why we didn't have to check for !volumePrefersRemote here. 
Something like 
```
This loop is calculating the total number of volumes that are available on one node.
This is used to decide at a later step if the node can be used to strictly enforce hyperconvergence 
based on preferLocalNodeOnly. preferLocalNodeOnly doesn't apply to volumes with NeedsAntiHyperconvergence set to true. So  an explicit volumePrefersRemoteOnly check is not necessary.
```